cosmos/operators/local.py

```diff
@@ -469,20 +488,33 @@ def get_datasets(self, source: Literal["inputs", "outputs"]) -> list[Dataset]:
             )
         return datasets
 
-    def register_dataset(self, new_inlets: list[Dataset], new_outlets: list[Dataset]) -> None:
+    def register_dataset(self, new_inlets: list[Dataset], new_outlets: list[Dataset], context: Context) -> None:
         """
-        Register a list of datasets as outlets of the current task.
+        Register a list of datasets as outlets of the current task, when possible.
+
         Until Airflow 2.7, there was not a better interface to associate outlets to a task during execution.
+        This works before Airflow 2.10 with a few limitations, as described in the ticket:
+        https://github.com/astronomer/astronomer-cosmos/issues/522
+
+        In Airflow 2.10.0 and 2.10.1, we are not able to test Airflow DAGs powered with DatasetAlias.
+        https://github.com/apache/airflow/issues/42495
```

Comment:
Should we mention somewhere in the docstring that we support the power of DatasetAlias in further versions?